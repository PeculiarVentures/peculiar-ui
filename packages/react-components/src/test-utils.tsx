import React from 'react';
import {
  render,
  screen,
  fireEvent,
  waitFor,
  renderHook,
  act,
} from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { ThemeProvider } from './styles/theme_provider';

const renderWithWrapper = (
  ui: React.ReactElement,
) => (
  render(
    ui,
    {
      wrapper: ThemeProvider,
    },
  )
);

export {
  userEvent,
  fireEvent,
  renderWithWrapper,
  screen,
  waitFor,
  renderHook,
  act,
};
