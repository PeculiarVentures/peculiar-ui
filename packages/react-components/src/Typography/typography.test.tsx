import React from 'react';
import { renderWithWrapper as render } from '../test-utils';
import { Typography } from './index';

describe('<Typography />', () => {
  it('should render with default styles', () => {
    const { asFragment } = render(
      <Typography>
        Typography message
      </Typography>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass className', () => {
    const { asFragment } = render(
      <Typography className="my-class-name">
        Typography message
      </Typography>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should pass "component"', () => {
    const { asFragment } = render(
      <Typography component="div">
        Typography message
      </Typography>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  describe('should render correct colors', () => {
    const colors: Array<React.ComponentProps<typeof Typography>['color']> = [
      'primary',
      'secondary',
      'wrong',
      'attention',
      'extra-1',
      'extra-2',
      'white',
      'black',
      'gray-1',
      'gray-2',
      'gray-3',
      'gray-4',
      'gray-5',
      'gray-6',
      'gray-7',
      'gray-8',
      'gray-9',
      'gray-10',
      'inherit',
    ];

    colors.forEach((color) => {
      it(`color: "${color}"`, () => {
        const { asFragment } = render(
          <Typography color={color}>
            Typography message
          </Typography>,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });

  describe('should render correct variants', () => {
    const variants: Array<React.ComponentProps<typeof Typography>['variant']> = [
      'h1',
      'h2',
      'h3',
      'h4',
      'h5',
      's1',
      's2',
      'b1',
      'b2',
      'b3',
      'btn1',
      'btn2',
      'c1',
      'c2',
    ];

    variants.forEach((variant) => {
      it(`variant: "${variant}"`, () => {
        const { asFragment } = render(
          <Typography variant={variant}>
            Typography message
          </Typography>,
        );

        expect(asFragment()).toMatchSnapshot();
      });
    });
  });
});
