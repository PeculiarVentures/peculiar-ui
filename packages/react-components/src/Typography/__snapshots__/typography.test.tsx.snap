// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Typography /> > should pass "component" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<div
    class="emotion-0"
  >
    Typography message
  </div>
</DocumentFragment>
`;

exports[`<Typography /> > should pass className 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="my-class-name emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "attention" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-attention);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "black" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "extra-1" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-extra-1);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "extra-2" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-extra-2);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-1" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-1);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-2" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-2);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-3" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-3);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-4" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-4);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-5" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-5);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-6" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-6);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-7" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-7);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-8" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-8);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-9" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-9);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "gray-10" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-gray-10);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "inherit" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: inherit;
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "primary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-primary);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "secondary" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-secondary);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "white" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-white);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct colors > color: "wrong" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-wrong);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "b1" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "b2" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b2-weight);
  font-size: var(--pv-text-b2-size);
  line-height: var(--pv-text-b2-height);
  letter-spacing: var(--pv-text-b2-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "b3" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b3-weight);
  font-size: var(--pv-text-b3-size);
  line-height: var(--pv-text-b3-height);
  letter-spacing: var(--pv-text-b3-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "btn1" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-btn1-weight);
  font-size: var(--pv-text-btn1-size);
  line-height: var(--pv-text-btn1-height);
  letter-spacing: var(--pv-text-btn1-spacing);
}

<span
    class="emotion-0"
  >
    Typography message
  </span>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "btn2" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-btn2-weight);
  font-size: var(--pv-text-btn2-size);
  line-height: var(--pv-text-btn2-height);
  letter-spacing: var(--pv-text-btn2-spacing);
}

<span
    class="emotion-0"
  >
    Typography message
  </span>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "c1" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-c1-weight);
  font-size: var(--pv-text-c1-size);
  line-height: var(--pv-text-c1-height);
  letter-spacing: var(--pv-text-c1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "c2" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-c2-weight);
  font-size: var(--pv-text-c2-size);
  line-height: var(--pv-text-c2-height);
  letter-spacing: var(--pv-text-c2-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "h1" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-h1-weight);
  font-size: var(--pv-text-h1-size);
  line-height: var(--pv-text-h1-height);
  letter-spacing: var(--pv-text-h1-spacing);
}

<h1
    class="emotion-0"
  >
    Typography message
  </h1>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "h2" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-h2-weight);
  font-size: var(--pv-text-h2-size);
  line-height: var(--pv-text-h2-height);
  letter-spacing: var(--pv-text-h2-spacing);
}

<h2
    class="emotion-0"
  >
    Typography message
  </h2>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "h3" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-h3-weight);
  font-size: var(--pv-text-h3-size);
  line-height: var(--pv-text-h3-height);
  letter-spacing: var(--pv-text-h3-spacing);
}

<h3
    class="emotion-0"
  >
    Typography message
  </h3>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "h4" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-h4-weight);
  font-size: var(--pv-text-h4-size);
  line-height: var(--pv-text-h4-height);
  letter-spacing: var(--pv-text-h4-spacing);
}

<h4
    class="emotion-0"
  >
    Typography message
  </h4>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "h5" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-h5-weight);
  font-size: var(--pv-text-h5-size);
  line-height: var(--pv-text-h5-height);
  letter-spacing: var(--pv-text-h5-spacing);
}

<h5
    class="emotion-0"
  >
    Typography message
  </h5>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "s1" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-s1-weight);
  font-size: var(--pv-text-s1-size);
  line-height: var(--pv-text-s1-height);
  letter-spacing: var(--pv-text-s1-spacing);
}

<h6
    class="emotion-0"
  >
    Typography message
  </h6>
</DocumentFragment>
`;

exports[`<Typography /> > should render correct variants > variant: "s2" 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-s2-weight);
  font-size: var(--pv-text-s2-size);
  line-height: var(--pv-text-s2-height);
  letter-spacing: var(--pv-text-s2-spacing);
}

<h6
    class="emotion-0"
  >
    Typography message
  </h6>
</DocumentFragment>
`;

exports[`<Typography /> > should render with default styles 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  color: var(--pv-color-black);
  font-weight: var(--pv-text-b1-weight);
  font-size: var(--pv-text-b1-size);
  line-height: var(--pv-text-b1-height);
  letter-spacing: var(--pv-text-b1-spacing);
}

<p
    class="emotion-0"
  >
    Typography message
  </p>
</DocumentFragment>
`;
